### PR TITLE
Refresh new command line interface for turnkeyml

### LINKED ISSUE #151: Refresh: New command-line interface for TurnkeyML
Turnkey commands will be entered into the CLI as an explicit sequence of Stages, as opposed to a named Sequence. Each Stage can have its own arguments.

Old way: `turnkey --all --args --are --global --sequence optimize-onnx`

New way: `turnkey --global --args Stage1 --stage1 --args Stage 2 --stage2 --args etc...`

### PR DESCRIPTION
**Pull request by jeremyfowers**
Closes #151 

# Note to Reviewers

Please note: I do not expect all aspects of the design here to be final. In addition to the other PRs to come in the refresh, I am keeping a list of design nits here that we may choose to address before the refresh is over: #168 

# Major Changes

- The `turnkey` CLI has been completely replaced.
- The `--sequence` argument has been removed and sequences are no longer auto-selected for the user.
  - Model Intake has been greatly simplified as a result; sequences.py has been removed.
  - Users must explicitly call the stages they want to run, for example `turnkey -i example.py export-pytorch`
  - Discovery and Benchmarking are implicit stages that run at the beginning and end of every sequence, respectively, They will be made explicit in future PRs.
      - The arguments space is messy right now because Discovery and Benchmarking arguments are mixed in with the global arguments. This will be fixed in future PRs as well.
- All non-benchmarking `turnkey` commands are now `ManagementTool`s now; these functionally similar `Stage`s, except they are not meant to be sequenced together.
  - This includes `turnkey cache`, `turnkey models`, and `turnkey version`
  - `turnkey cache benchmark` is now a `ManagementTool`, which means it will be very easy to tease out into a plugin later.
  - **New feature:** `ManagementTools` can be installed by plugins, so external/private repos can extend the functionality (e.g., add their own version of a reporting tool)
- Plugins no longer install sequences; now they install individual stages, which can be sequenced any way the user likes.
  - The `example_sequence` plugin is now the `example_stage` plugin.
- The `Sequence` class no longer inherits `Stage`.
  - Nested `Sequences` are no longer allowed (nobody was doing this anyways)
  - Design significantly simplified as a result.
- Every `Stage` now has its own arguments parser, and stages can now take per-stage arguments.
  - For example: `turnkey -i example.py export-pytorch --opset 18`

### CODE REVIEW

File: src/turnkeyml/analyze/script.py [hunk 1]
```diff
@@ -522,6 +518,8 @@ def explore_invocation(
 
         _store_traceback(invocation_info)
 
+        raise e
```
Comment (jeremyfowers) [inline src/turnkeyml/analyze/script.py:521]: Remove

Comment (jeremyfowers) [reply src/turnkeyml/analyze/script.py:521]: Or wrap with TURNKEY_DEBUG

File: src/turnkeyml/common/management_tools.py [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+import turnkeyml.common.printing as printing
+from turnkeyml.version import __version__ as turnkey_version
+from turnkeyml.cli.report import summary_spreadsheets
+
+
+class ManagementTool(abc.ABC):
+    """
+    Intended for management functions, such as managing the cache
+    or printing the version number.
+    """
```
Comment (jeremyfowers) [inline src/turnkeyml/common/management_tools.py:16]: ADD `unique_name`

File: src/turnkeyml/common/management_tools.py [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+        return parser
+
+    def run(self, _, quiet: bool = False):
+        if quiet:
+            print(fs.MODELS_DIR)
+        else:
+            printing.log_info(f"The models directory is: {fs.MODELS_DIR}")
+
+
+class Report(ManagementTool):
```
Comment (jeremyfowers) [inline src/turnkeyml/common/management_tools.py:246]: move to report.py

File: test/analysis.py [hunk 4]
```diff
@@ -165,6 +165,7 @@ def run_cli(args):
 
 def run_analysis(args):
     output = run_cli(args)
+    print(output)
```
Comment (jeremyfowers) [inline test/analysis.py:168]: remove

**Review by danielholanda — APPROVED** [review]

**Review by ramkrishna2910 — APPROVED** [review]

**Pull request merged**